core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -242,21 +260,28 @@ class BpmBillingProjectLifecycle(
           )
         case _ => ()
       }
-      _ <- billingRepository.getLandingZoneId(projectName).flatMap {
+      (jobControlId, eventType) <- billingRepository.getLandingZoneId(projectName).map {
         case Some(landingZoneId) =>
-          cleanupLandingZone(UUID.fromString(landingZoneId), projectName, ctx)
+          val result = cleanupLandingZone(UUID.fromString(landingZoneId), ctx)
+          (UUID.fromString(result.getJobReport.getId), AzureBillingProjectDelete)
         case None =>
-          logger.warn(s"Deleting BPM-backed billing project ${projectName}, but no associated landing zone to delete")
-          Future.successful()
-      }
-      _ <- billingRepository.getBillingProfileId(projectName).flatMap {
-        case Some(billingProfileId) =>
-          cleanupBillingProfile(UUID.fromString(billingProfileId), projectName, ctx)
-        case None =>
-          logger.warn(
-            s"Deleting BPM-backed billing project ${projectName}, but no associated billing profile record to delete"
-          )
-          Future.successful()
+          logger.warn(s"Deleting BPM-backed billing project $projectName, but no associated landing zone to delete")
+          (UUID.randomUUID(), OtherBpmBillingProjectDelete)
       }
-    } yield {}
+    } yield (jobControlId, eventType)
+
+  override def finalizeDelete(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext)(implicit
+    executionContext: ExecutionContext
+  ): Future[Unit] = for {
+    billingProfileId <- billingRepository.getBillingProfileId(projectName)
+    _ <- billingProfileId match {
+      case Some(id) => cleanupBillingProfile(UUID.fromString(id), projectName, ctx)
+      case None =>
+        logger.warn(
+          s"Deleting BPM-backed billing project $projectName, but no associated billing profile record was deleted"
```

Comment:
I think it would be good to make it clear that the billing profile record wasn't deleted because none existed-- this message makes it seem like it could have been there, but we didn't delete it.

Reply:
I was using the existing message - you're right though: updated message for clarity.